torchtext/models/roberta/modules.py

```diff
@@ -282,15 +284,25 @@ def forward(self, tokens: torch.Tensor) -> List[torch.Tensor]:
 
         encoded = padded_embedded.transpose(0, 1)
 
-        states = [encoded]
+        if self.return_all_layers:
+            states = [encoded]
 
-        for layer in self.layers:
-            encoded = layer(encoded, padding_mask)
-            states.append(encoded)
+            for layer in self.layers:
+                encoded = layer(encoded, padding_mask)
```

Comment:
One nit for readability is to consider moving this for loop outside the if statements, since it is the same in both cases. Either way is fine though :)

Reply:
Actually we are also storing all the layer states when return_all_layers is set True. If for loop is moved outside, then I would need to bring the if loop inside the for loop :).